app/controllers/mixins/actions/vm_actions/transform.rb

```diff
@@ -17,7 +17,8 @@ def vm_transform
             @right_cell_text = _("Transform VMs to RHV")
             simple_dialog_initialize(
               dialog.resource_actions.first,
-              :header => @right_cell_text
+              :header => @right_cell_text,
```

Comment:
@smelamud This doesn't look right.
First of this is a Provider Action not a VM action. When the UI calls the controller with no vms selected the URL looks like
http://localhost:3000/ems_infra/10000000000001?display=vms

We get the params as <ActionController::Parameters {"pressed"=>"vm_transform", "controller"=>"ems_infra", "action"=>"button", "id"=>"10000000000001"} permitted: false>

So without the context vm or provider we can't be calling vm_transform and expecting that the ID is the ID of a VM when in reality it is the ID of the provider.

I would think we should have 2 separate actions
vm_transform which works with a single/multiple vm's, the target would be the vm object

mass_transform which applies to all tagged VMs in the provider, the target object here is the provider which the controller is passing it in.

So the target needs to be set appropriately based on the context.

Currently this line https://github.com/ManageIQ/manageiq-ui-classic/blob/d4e9e12960a17a3a5104755462aaea41b9b94126/app/controllers/mixins/actions/vm_actions/transform.rb#L7 is wrong without the context. It gives  arbitrary vm object for which there is a different BZ open.

Reply:
If `mass_transform` is a provider action, what module/class I need to put it into?